llama_index/indices/multi_modal/base.py

```diff
@@ -169,6 +182,12 @@ def _get_node_with_embedding(
             embedding = id_to_embed_map[node.node_id]
             result = node.copy()
             result.embedding = embedding
+            if is_image and id_to_text_embed_map:
+                text_embedding = id_to_text_embed_map[node.node_id]
+                result.text_embedding = text_embedding
```

Comment:
do we have text_embedding for node?

Reply:
yea so result is a copy of the `node` (see L. 183) and if `is_image` is `True`, then `node` and thus `result` are `ImageNode`. And, I just added `text_embedding` as a field to `ImageNode`.